classes/notification/PKPNotificationOperationManager.inc.php

```diff
@@ -392,6 +412,69 @@ protected function sendNotificationEmail($request, $notification, ?int $contextI
 			}
 		}
 	}
+
+	/**
+	 * Creates and returns a unique string for the given notification, that will be encoded and validated against.
+	 * @param $notification Notification
+	 * @return string
+	 */
+	function createUnsubscribeUniqueKey($notification) {
+		$uniqueKey = 'unsubscribe' . '-' . $notification->getContextId() . '-' . $notification->getUserId() . '-' . $notification->getId();
+
+		return $uniqueKey;
+	}
+
+	/**
+	 * Creates and returns an encoded token that will be used to validate an unsubscribe url.
+	 * @param $notification Notification
+	 * @return string
+	 */
+	function createUnsubscribeToken($notification) {
+		$encodeString = $this->createUnsubscribeUniqueKey($notification);
+
+		$secret = Config::getVar('security', 'signed_page_key_secret', '');
+		$jwt = '';
+		if ($secret !== '') {
+			$jwt = JWT::encode(json_encode($encodeString), $secret, 'HS256');
+		}
+
+		return $jwt;
+	}
+
+	/**
+	 * The given notification is validated against the requested token.
+	 * @param $token string
+	 * @param $notification Notification
+	 * @return bool
+	 */
+	function validateUnsubscribeToken($token, $notification) {
+		$encodeString = $this->createUnsubscribeUniqueKey($notification);
+
+		$secret = Config::getVar('security', 'signed_page_key_secret', '');
+		$jwt = '';
+		if ($secret !== '') {
+			$jwt = json_decode(JWT::decode($token, $secret, array('HS256')));
+		}
+
+		if ($jwt == $encodeString) {
```

Comment:
If the `signed_page_key_secret` variable in `config.inc.php` is not set -- which is the default -- I think this will result in all tokens being accepted!

Reply:
In fact, if the `signed_page_key_secret` variable is not set, the token will not be created, thus no unsubscribe url will be accepted. Should we add a default value, or follow the `api_key_secret` and leave it empty? (or use the the  `api_key_secret` for this also?)

Reply:
Gotcha thanks -- let's re-use `api_key_secret`.

Reply:
Changed to `api_key_secret`. Still it is empty, so no unsubscribe url will be accepted.